lib/chef/provider/package/rpm.rb

```diff
@@ -60,8 +60,7 @@ def load_current_resource
             end
 
             Chef::Log.debug("#{@new_resource} checking rpm status")
-            status = shell_out!("rpm -qp --queryformat '%{NAME} %{VERSION}-%{RELEASE}\n' #{@new_resource.source}")
-            status.stdout.each_line do |line|
+            shell_out!("rpm -qp --queryformat '%{NAME} %{VERSION}-%{RELEASE}\n' #{@new_resource.source}").stdout.each_line do |line|
```

Comment:
We cannot use shell_out! and need to use simple shell_out here since load_current_resource will start throwing exceptions and we wont be able to go ahead with fresh rpm installation. The specs fails on AIX as well as centos with shell_out!().